src/Core/Services/MetadataProviders/CosmosSqlMetadataProvider.cs

```diff
@@ -222,6 +222,16 @@ private void ParseSchemaGraphQLFieldsForJoins()
                 }
 
                 string entityType = field.Type.NamedType().Name.Value;
+
+                // If the entity is not present in the runtime config, throw an exception as we are expecting all the entities to be present in the runtime config.
+                if (!_runtimeConfig.Entities.ContainsKey(entityType))
+                {
+                    throw new DataApiBuilderException(
+                        message: $"The entity '{entityType}' was not found in the runtime config.",
+                        statusCode: System.Net.HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
+                }
```

Comment:
Just curious, where was the exception being thrown before this change? Please mention this in PR description.

Reply:
mentioned the PR in description